petastorm/tests/test_spark_dataset_converter.py

```diff
@@ -42,16 +46,24 @@ def test_primitive(self):
                 # Now we only have one batch.
             for i in range(converter.dataset_size):
                 for col in df.schema.names:
-                    self.assertEquals(getattr(ts, col)[i], expected_df[i][col])
+                    actual_ele = getattr(ts, col)[i]
+                    expected_ele = expected_df[i][col]
+                    if type(actual_ele) == bytes:
+                        actual_ele = actual_ele.decode()
```

Comment:
We should directly compare "bytes" type in python. The decode is used when there's encoding like utf8. Here the "Binary" type in spark support arbitrary binary content, such as raw file content bytes.

Reply:
For `StringType` column, the `actual_ele` will be `b'hello'` (bytes: 5) while `expected_ele` is `'hello'` (str).

Reply:
I summarized the mapping in the table in the PR description. The `Evaluated tensor element type` is the type of `actual ele` and the `rows=df.collect()` is the type of `expected_ele`.

Reply:
For BinaryType, I cast `actual_ele` from `bytes` to `bytearray`. For StringType, I decode `actual_ele` from `bytes` to `str` using default decode=`ascii`.

Reply:
oh, looks good. But if it is string type, the `getattr(ts, col)[i]` should already been a str object, I guess tensorflow already decode it ?

Reply:
Oh, you're right, tf.string type element always is b'xxx' type so need decode.